File: anylang/go/leetcode/1534.统计好三元组.go
package leetcode

import (
	"fmt"
)

func abs(num int) int {
	if num > 0 {
		return num
	}
	return -num
}

func countGoodTriplets(arr []int, a int, b int, c int) int {
	length := len(arr)
	rst := 0
	for i := 0; i < length; i++ {
		for j := i + 1; j < length; j++ {
			for k := j + 1; k < length; k++ {
				if abs(arr[i]-arr[j]) <= a && abs(arr[j]-arr[k]) <= b && abs(arr[i]-arr[k]) <= c {
					rst++
				}
			}
		}
	}

	return rst
}

func CountGoodTriplets() {
	arr := []int{3, 0, 1, 1, 9, 7}
	// arr := []int{1, 1, 2, 2, 3}
	fmt.Println("countGoodTriplets({3,0,1,1,9,7}, 7, 2, 3)", countGoodTriplets(arr, 7, 2, 3))
}

File: anylang/clang/1534.统计好三元组.c
#include <stdlib.h>
#include <stdio.h>

int countGoodTriplets(int* arr, int arrSize, int a, int b, int c) {
    int rst = 0;
    for (int i = 0; i < arrSize; i++) {
        for (int j = i + 1; j < arrSize; j++) {
            for (int k = j + 1; k < arrSize; k++) {
                if (abs(arr[i] - arr[j]) <= a && abs(arr[j] - arr[k]) <= b && abs(arr[i] - arr[k]) <= c) {
                    rst++;
                }
            }
        }
    }

    return rst;
}

int main() {
    int arr[] = {3, 0, 1, 1, 9, 7};
    // arr := []int{1, 1, 2, 2, 3}
    printf("countGoodTriplets({3,0,1,1,9,7}, 7, 2, 3) %d", countGoodTriplets(arr, 6, 7, 2, 3));
    return 0;
}
